src/briefcase/integrations/java.py

```diff
@@ -51,6 +51,30 @@ def OpenJDK_download_url(self):
             f"{self.JDK_RELEASE}_{self.JDK_BUILD}.{extension}"
         )
 
+    @classmethod
+    def version_from_path(cls, tools: ToolCache, java_path: str | Path) -> str:
+        """Return a JDK's version from a path by running ``<java_path>/bin/javac``.
+
+        This will fail if the path contains a JRE instead of a JDK.
+
+        Several exceptions can be raised for issues:
+         - OSError - the ``javac`` executable doesn't exist
+         - CalledProcessError - ``javac`` returned a non-zero value
+         - IndexError - unparsable version value from ``javac``
+
+        :param tools: ToolCache of available tools
+        :param java_path: File path to a candidate JDK install
+        :return: JDK release version; e.g. "17.0.7"
+        """
+        output = tools.subprocess.check_output(
+            [
+                os.fsdecode(Path(java_path) / "bin" / "javac"),
+                "-version",
+            ],
+        )
+        # This should be a string of the form "javac 17.0.7\n"
```

Comment:
This comment is misleading (/stale) - the return value doesn't include `javac `.

Reply:
Are you seeing this somewhere in particular? I'm getting `javac 17.0.7` as the output.

Reply:
Possibly confusing placement/wording then. I read "this" as applying to return value, not the input. The input definitely has `javac `; but the return value is just 17.0.7.